### PR TITLE
Split log line generation from message search text

### LINKED ISSUE #4697: Usernames without nicknames in logs are duplicated
### Checklist

- [X] I'm reporting a problem with Chatterino
- [X] I've verified that I'm running **the most** recent nightly build or stable release
- [X] I've looked for my problem on the [wiki](https://wiki.chatterino.com/Help/)
- [X] I've searched the [issues and pull requests](https://github.com/Chatterino/chatterino2/issues?q=) for similar looking reports

### Describe your issue

If a user doesn't have a nickname set, their username is duplicated in chat logging. See example:

```
# Current behaviour:
[16:07:31] praxis099  praxis099: bag for cup but no cup
[16:07:44] beyson2301(龍亨街市) 龍亨街市 beyson2301: BBQ in 31 degree celsius monkaS

# With nickname set ("streamelements" set to "se"):
[18:47:29] se  streamelements: Posture check SwiftRage

# Previous behaviour: 
[15:40:19]  koromochii: KEKW
[15:40:20] ダスビーター das_beater: POGSPIN
```

Appears to be an unintentional side-effect due to the way https://github.com/Chatterino/chatterino2/pull/4663 was implemented (logging uses the search text of the message). IMO, nicknames don't really belong in chatlogs, but if we decide to include them, they probably shouldn't double-up like that.

### Screenshots

_No response_

### OS and Chatterino Version

Chatterino Nightly 2.4.4 (commit 53b044c4) built on 2023-06-18 with Qt 6.5.0, MSVC 193632532 Running on Windows 11 Version 23H2, kernel: 10.0.25381

### PR DESCRIPTION
**Pull request by Mm2PL**
# Description

Closes #4697

The line string is now generated in logging code and search text isn't used.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -31,6 +31,7 @@
 - Bugfix: Fixed highlights sometimes not working after changing sound device, or switching users in your operating system. (#4729)
 - Bugfix: Fixed key bindings not showing in context menus on Mac. (#4722)
 - Bugfix: Fixed tab completion rarely completing the wrong word. (#4735)
+- Bugfix: Fixed duplicate username in logs (#4742)
```
Comment (Felanbird) [inline CHANGELOG.md:34]: can be stuck on the `nicknames are now taken into consideration` changelog

**Review by Felanbird — APPROVED** [review]
image

**Comment by pajlada** [comment]
This still generates duplicate usernames in logs

Without this PR:
```
[13:32:06] pajlada  pajlada: xd
[13:32:19] pieladder  pajlada: xbcvbcvb
```

With this PR:
```
[13:28:44] pajlada  pajlada: b
[13:29:02] pieladder  pajlada: c
```

**Review by pajlada — APPROVED** [review]
Works as expected

After this PR:
```
[13:49:58] pajlada: asd << without nickname
[13:50:00] 테스트계정420 testaccount_420: xd
[13:50:18] pajlada: xddddd << with nickname
```

**Comment by pajlada** [comment]
@Mm2PL has mentioned that this will make a change in how usernames are logged; the previous way was to have two spaces if no localized username was found, this PR changes this with my additional commits

**Pull request merged**